server/emails/templates/BaseEmail.tsx

```diff
@@ -80,6 +81,21 @@ export default abstract class BaseEmail<T extends EmailProps, S = any> {
       Metrics.increment("email.sent", {
         templateName,
       });
+
+      // @ts-expect-error does not exist on type T & S
+      if (data.notificationId) {
+        await Notification.update(
```

Comment:
I'm worried that this within the try/catch could cause an email to be sent twice if it fails, better to move outside the try and fail with a Logger

Reply:
Thanks for spotting this :pray: